torch/distributions/poisson.py

```diff
@@ -0,0 +1,53 @@
+from numbers import Number
+
+import torch
+from torch.autograd import Variable
+from torch.distributions import constraints
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all
+
+
+def _poisson(_lambda):
+    if not isinstance(_lambda, Variable):
+        return torch.poisson(Variable(_lambda)).data
+    return torch.poisson(_lambda)
+
+
+class Poisson(Distribution):
+    r"""
+    Creates a Poisson distribution parameterized by `_lambda`, the rate parameter.
```

Comment:
nit: preceding underscore usually denotes a private variable. It's probably safer to call this `lam` or `lambda_`. Alternatively it would be nice to call it `rate` which is both semantically meaningful and improves compatibility with tensorflow distributions.

Reply:
For example `self._lambda` would be the obvious member name but that looks private. Instead `self.lambda_` or `self.rate` are clearly public.

Reply:
Both sound good to me. Let's use `rate` since that's more evocative? I.e. like `mean`/`std` in `Normal` already, and matches up with `Exponential`, which is the right idea.